src/ResourceManager/Compute/Commands.Compute/Generated/ComputeAutomationBaseCmdlet.cs

```diff
@@ -300,13 +300,23 @@ public static string GetResourceGroupName(string resourceId)
             return m.Success ? m.Groups["rgname"].Value : null;
         }
 
-        public static string GetResourceName(string resourceId, string resourceName)
+        public static string GetResourceName(string resourceId, string resourceName, string instanceName = null)
         {
             if (string.IsNullOrEmpty(resourceId)) { return null; }
-            Regex r = new Regex(@"(.*?)/" + resourceName + @"/(?<rgname>\S+)", RegexOptions.IgnoreCase);
+            Regex r = (instanceName == null)
+                      ? new Regex(@"(.*?)/" + resourceName + @"/(?<rgname>\S+)", RegexOptions.IgnoreCase)
```

Comment:
it seems the three reg exes here share the same prefix, i.e. : 
 ```
@"(.*?)/" + resourceName + @"/(?<rgname>\S+)"
```
Can you move it to a common const that gets consumed in all the three locations??